pmd-core/src/main/java/net/sourceforge/pmd/lang/ast/impl/javacc/CharStream.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.ast.impl.javacc;
+
+
+import java.io.EOFException;
+
+import net.sourceforge.pmd.lang.ast.impl.javacc.JavaccTokenDocument.TokenDocumentBehavior;
+import net.sourceforge.pmd.lang.document.Chars;
+import net.sourceforge.pmd.lang.document.FileLocation;
+import net.sourceforge.pmd.lang.document.TextDocument;
+import net.sourceforge.pmd.lang.document.TextRegion;
+
+/**
+ * PMD flavour of character streams used by JavaCC parsers.
+ */
+public final class CharStream {
+
+    private static final EOFException EOF = new EOFException();
```

Comment:
Fixed in f3dbdb700cc0ef592b31cab299a38c08ba3f6d4d

Reply:
This exception is caught immediately by Javacc and ignored, and is thrown in only one place (but frequently)... The stack trace is costly to build and not useful in this case IMO

Reply:
I'd rather wait with such optimizations until we know, that it's necessary. And then we need to add an explaining comment, otherwise I'll remove this "optimization" again :smile:

Reply:
Fair enough :)